### ISSUE TITLE
code_on_top_scope warns about node definitions

### ISSUE THREAD

**Issue by paran1** [rodjek/puppet-lint#580]
Example input
```
$ cat node.pp 
node 'faran'
{
}
```
Expected: no warnings

Actual result:
```
$ puppet-lint node.pp 
WARNING: code outside of class or define block - node on line 1
WARNING: code outside of class or define block - faran on line 1
WARNING: code outside of class or define block - { on line 2
WARNING: code outside of class or define block - } on line 3
$ 
```

Tested with version 3.0.0 and latest master (041b4303108309159e181092c883d2e6b8413649) with same result.

**Comment by paran1** [comment]
I just realized that 3.0.0 is a branch rather than a tagged version, so please ignore that. The code_on_top_scope check is not in any released version.

**Comment by rnelson0** [comment]
This is by design; ALL code on the top scope is flagged when this check is left enabled.

**Comment by paran1** [comment]
Yes, and that design is wrong. Node definitions are supposed to be in the main manifests, top scope.

**Comment by rnelson0** [comment]
In that case, it would be appropriate to disable the check for the file with a flag or control comment. A node statement anywhere other than `manifests/site.pp` is incorrect, but puppet-lint is looking at the content rather than whether the file is the site manifest.

**Comment by paran1** [comment]
NNode definitions is a standard language feature, not deprecated by the style guide. I expect to be able to run puppet-lint on such manifests without me having to add control comments or use flags.

Also, node definitions can be in any file name, as the site manifests might be specified using a manifests directory.
